.github/workflows/kubectl-az.yml

```diff
@@ -0,0 +1,199 @@
+name: Microsoft Azure CLI kubectl plugin CI
+env:
+  GO_VERSION: 1.17 # TODO: Update
+  AZURE_PREFIX: kubectl-az-ci
+concurrency:
+  # Only one workflow can run at a time unless
+  # we create a new AKS cluster per github_ref (branch)
+  group: kubectl-az-ci
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build kubectl-az
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ linux, darwin, windows ]
+        arch: [ amd64, arm64 ]
+        exclude:
+          - os: windows
+            arch: arm64
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          check-latest: true
+      - name: Cache Go
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Build and generate tarball
+        run: |
+          target=kubectl-az-${{ matrix.os }}-${{ matrix.arch }}
+
+          make $target
+
+          binary_name=kubectl-az
+          if [ ${{ matrix.os }} = "windows" ]; then
+            binary_name=kubectl-az.exe
+          fi
+
+          # Prepare binary as artifact, it will be used by other jobs
+          mv $target $binary_name
+          tar --sort=name --owner=root:0 --group=root:0 \
+            -czf ${target}.tar.gz \
+            $binary_name LICENSE
+      - name: Add kubectl-az-${{ matrix.os }}-${{ matrix.arch }}.tar.gz as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: kubectl-az-${{ matrix.os }}-${{ matrix.arch }}
+          path: kubectl-az-${{ matrix.os }}-${{ matrix.arch }}.tar.gz
+
+  unit-tests:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          check-latest: true
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Cache Go
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Run tests
+        run: make unit-test
+
+  create-aks-cluster:
```

Comment:
AFAIU, they are separated because they need to be run only once and not for each element in the `integration-tests`'s matrix:

https://github.com/Azure/kubectl-az/blob/f778a8e8e27cd4bd78bb7463137f8b5077278a8a/.github/workflows/kubectl-az.yml#L145

Reply:
Makes sense. Was thinking about ADO pipelines :sweat_smile: